stagemonitor-web-servlet/src/test/java/org/stagemonitor/web/servlet/eum/ClientSpanServletTest.java

```diff
@@ -0,0 +1,273 @@
+		// Given
+		MockHttpServletRequest mockHttpServletRequest = new MockHttpServletRequest();
+		mockHttpServletRequest.setParameter("ty", "pl");
+		mockHttpServletRequest.setParameter("r", "1496751574200");
+		mockHttpServletRequest.setParameter("u", "http://localhost:9966/petclinic/");
+		mockHttpServletRequest.setParameter("m_user", "tom.mason@example.com");
+		mockHttpServletRequest.setParameter("ts", "-197");
+		mockHttpServletRequest.setParameter("d", "518");
+		mockHttpServletRequest.setParameter("t_unl", "0");
+		mockHttpServletRequest.setParameter("t_red", "0");
+		mockHttpServletRequest.setParameter("t_apc", "5");
+		mockHttpServletRequest.setParameter("t_dns", "0");
+		mockHttpServletRequest.setParameter("t_tcp", "0");
+		mockHttpServletRequest.setParameter("t_req", "38");
+		mockHttpServletRequest.setParameter("t_rsp", "4");
+		mockHttpServletRequest.setParameter("t_pro", "471");
+		mockHttpServletRequest.setParameter("t_loa", "5");
+		mockHttpServletRequest.setParameter("t_fp", "151");
+
+		// TODO, ignore for now
+		mockHttpServletRequest.setParameter("k", "someKey"); // not necessary
+		mockHttpServletRequest.setParameter("t", "a6b5fd025be24191"); // trace id -> opentracing does not specify this yet
+		mockHttpServletRequest.setParameter("bt", "null"); // not sure if necessary
+		mockHttpServletRequest.setParameter("res", "{\"http://localhost:9966/petclinic/\":{\"webjars/\":{\"bootstrap/2.3.0/\":{\"css/bootstrap.min.css\":[\"-136,10,2,1,105939\"],\"img/glyphicons-halflings.png\":[\"-48,0,4,1,12799\"]},\"jquery\":{\"/2.0.3/jquery.js\":[\"-136,0,3,1,242142\"],\"-ui/1.10.3/\":{\"ui/jquery.ui.\":{\"core.js\":[\"-136,0,3,1,8198\"],\"datepicker.js\":[\"-136,0,3,1,76324\"]},\"themes/base/jquery-ui.css\":[\"-136,8,2,1,32456\"]}}},\"resources/\":{\"css/petclinic.css\":[\"-136,8,2,1,243\"],\"images/\":{\"banner-graphic.png\":[\"-136,0,1,1,13773\"],\"pets.png\":[\"-136,0,1,1,55318\"],\"spring-pivotal-logo.png\":[\"-136,0,1,1,2818\"]}},\"stagemonitor/\":{\"public/static/\":{\"rum/boomerang-56c823668fc.min.js\":[\"-136,0,3,1,12165\"],\"eum.debug.js\":[\"-32,12,3,3,23798\"]},\"static/stagemonitor\":{\".png\":[\"-136,16,1,3,1694\"],\"-modal.html\":[\"-33,9,5,3,10538\"]}}}}");
+
+		// When
+		servlet.doPost(mockHttpServletRequest, new MockHttpServletResponse());
+
+		// Then
+		assertSoftly(softly -> {
+			final List<MockSpan> finishedSpans = tracer.finishedSpans();
+			softly.assertThat(finishedSpans).hasSize(1);
+			MockSpan span = finishedSpans.get(0);
+			softly.assertThat(span.operationName()).isEqualTo("/petclinic/");
+			softly.assertThat(span.startMicros()).isEqualTo(TimeUnit.MILLISECONDS.toMicros(1496751574200L + -197L));
+			softly.assertThat(span.finishMicros()).isEqualTo(TimeUnit.MILLISECONDS.toMicros(1496751574200L + -197L + 518L));
+
+			softly.assertThat(span.tags())
+					.containsEntry("type", "pageload")
+					.doesNotContainKey("user") // TODO: test whitelisting of metatags
```

Comment:
Seems like the TODO is resolved by `testConvertWeaselTraceToStagemonitorTrace_withMetadata`